### PR TITLE
Allow build-time configuration of NRF_MODEM_LIB_NET_IF_DOWN

### PR DESCRIPTION
**Pull request by glarsennordic**
This PR adds a KConfig option which allows the default value of the NRF_MODEM_LIB_NET_IF_DOWN option to be changed at build time.

Also, while I was in the area, I updated the flag overrides to use a newly available API

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -412,10 +412,14 @@ Modem libraries
...
 * :ref:`nrf_modem_lib_readme`:
 
-   * Added a mention about enabling TF-M logging while using modem traces in the :ref:`modem_trace_module`.
-   * Updated by renaming ``lte_connectivity`` module to ``lte_net_if``.
-     All related Kconfig options have been renamed accordingly.
-   * Changed the default value of the :kconfig:option:`CONFIG_NRF_MODEM_LIB_NET_IF_AUTO_START`, :kconfig:option:`CONFIG_NRF_MODEM_LIB_NET_IF_AUTO_CONNECT`, and :kconfig:option:`CONFIG_NRF_MODEM_LIB_NET_IF_AUTO_DOWN` Kconfig options from enabled to disabled.
+  * Added:
+
+    * A mention about enabling TF-M logging while using modem traces in the :ref:`modem_trace_module`.
+    * The :kconfig:option:`CONFIG_NRF_MODEM_LIB_NET_IF_DOWN_DEFAULT_LTE_DISCONNECT` option, allowing the :c:enumerator:`~nrf_modem_lib_net_if_down_options.NRF_MODEM_LIB_NET_IF_DOWN_LTE_DISCONNECT` ``conn_mgr`` iface option to be selected at build time.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:418]:
```suggestion
    * The :kconfig:option:`CONFIG_NRF_MODEM_LIB_NET_IF_DOWN_DEFAULT_LTE_DISCONNECT` option, allowing the ``conn_mgr`` iface option of the :c:enumerator:`NRF_MODEM_LIB_NET_IF_DOWN` enumerator to be selected at build time.
```
I may have messed up this totally. However, this needs a bit closer study. The enumerator was not rendered properly.

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by lemrey — COMMENTED** [review]
I was planning to do the same thing, but at the same time remove the capability for this configuration to be set at runtime. In my opinion, there are no use cases where the behavior of `lte_net_if_down` would be changed at runtime. At runtime, the firmware can call `lte_net_if_down` to bring the interface down and `conn_mgr_disconnect` to disconnect.

But I agree that it can be useful for the user to configure the behavior at at build time, so that it can be aligned with the behavior of other interfaces, for example for the purposes of achieving the same behavior of `_AUTO_DOWN`.

File: lib/nrf_modem_lib/lte_net_if/lte_net_if.c [hunk 2]
```diff
@@ -344,35 +344,28 @@ static void lte_net_if_init(struct conn_mgr_conn_binding *if_conn)
...
 	int ret;
 	int timeout = CONFIG_NRF_MODEM_LIB_NET_IF_CONNECT_TIMEOUT_SECONDS;
+	uint8_t down_mode;
 
 	net_if_dormant_on(if_conn->iface);
 
+	if (IS_ENABLED(NRF_MODEM_LIB_NET_IF_DOWN_DEFAULT_LTE_DISCONNECT)) {
+		/* It is impossible for this call to result in error. */
+		down_mode = NRF_MODEM_LIB_NET_IF_DOWN_LTE_DISCONNECT;
+		(void)conn_mgr_if_set_opt(if_conn->iface, CONN_MGR_IF_NO_AUTO_CONNECT,
```
[the remark this answers is not in the document]
Comment (glarsennordic) [reply lib/nrf_modem_lib/lte_net_if/lte_net_if.c:354]: Oh boy. I must have been tired yesterday to have missed this.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 1 above)
Comment (glarsennordic) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:418]: Hmm, I've had this exact syntax work over on the Zephyr documentation:

https://docs.zephyrproject.org/latest/connectivity/networking/conn_mgr/main.html#connection-persistence

```
Each iface also has a connection persistence setting that you can enable or disable by setting the :c:enumerator:`~conn_mgr_if_flag.CONN_MGR_IF_PERSISTENT` flag with :c:func:`conn_mgr_binding_set_flag`.
```

I'll dig into it

Comment (glarsennordic) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:418]: Ah, I spotted the problem I think. The `nrf_modem_lib_net_if_down_options` struct is inside a preprocessor guard that's missing `|| defined(__DOXYGEN__)`

**Comment by glarsennordic** [comment]
DNM until I get a chance to test this in some actual builds -- But reviews welcomed

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 1 above)
Comment (glarsennordic) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Yep, appears to work now

**Review by peknis — APPROVED** [review]

**Review by simensrostad — APPROVED** [review]
Thanks

**Pull request merged**